tests/console/repo_package_install.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: check SLE repositories, packages and installation status
+# Maintainer: Zaoliang Luo <zluo@suse.com>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+
+my @packages = (
+    {
+        name   => 'salt',
+        repo   => 'Basesystem_Module',
+        status => 'not installed'
+    });
+
+sub run {
+    foreach (@packages) {
+        #run zypper command to check repositories, packages and installation status
+        assert_script_run("zypper lr | grep '" . $_->{repo} . "'");
+        assert_script_run("zypper info '" . $_->{name} . "' | grep '" . $_->{repo} . "'");
+        assert_script_run("zypper info '" . $_->{name} . "' | grep '" . $_->{status} . "'");
+    }
```

Comment:
Instead of doing all that in individual calls I suggest `zypper search`. The whole hash section and `sub run` can look like this:

```perl
use utils 'zypper_call';

my %packages = (
    salt => {
        repo   => 'Basesystem',
        installed => 0
    });

for my $package (keys %packages) {
    my $args = $packages{$package}->{installed} ? '--installed-only' : '--non-installed-only';
    zypper_call('se ' . $args . ' --match-exact --details ' . $package . ' | grep ' . $packages{$package}->{repo});
}
```

Reply:
s/non-installed-only/not-installed-only/